core/slice.go

```diff
@@ -586,69 +587,70 @@ func (sl *Slice) computePendingHeader(localPendingHeaderWithTermini types.Pendin
 // updatePhCacheFromDom combines the recieved pending header with the pending header stored locally at a given terminus for specified context
 func (sl *Slice) updatePhCacheFromDom(pendingHeader types.PendingHeader, terminiIndex int, indices []int, reorg bool) error {
 
-	var localPendingHeader types.PendingHeader
 	hash := pendingHeader.Termini[terminiIndex]
 	localPendingHeader, exists := sl.phCache[hash]
 
 	if exists {
+		combinedPendingHeader := types.CopyHeader(localPendingHeader.Header)
 		for _, i := range indices {
-			localPendingHeader.Header = sl.combinePendingHeader(pendingHeader.Header, localPendingHeader.Header, i)
+			combinedPendingHeader = sl.combinePendingHeader(pendingHeader.Header, combinedPendingHeader, i)
 		}
-		localPendingHeader.Header.SetLocation(common.NodeLocation)
-		sl.phCache[hash] = localPendingHeader
+		combinedPendingHeader.SetLocation(common.NodeLocation)
+
+		sl.writeToPhCache(types.PendingHeader{Header: combinedPendingHeader, Termini: localPendingHeader.Termini}, false, TerminusIndex)
+		sl.pickPhCacheHead(reorg, types.PendingHeader{Header: combinedPendingHeader, Termini: localPendingHeader.Termini}, TerminusIndex)
 
-		if reorg {
-			sl.pendingHeaderHeadHash = hash
-		}
 		return nil
 	}
 	log.Warn("no pending header found for", "terminus", hash)
 	return errors.New("no pending header found in cache")
 }
 
-// writePhCache dom writes a given pendingHeaderWithTermini to the cache with the terminus used as the key.
-func (sl *Slice) writeToPhCache(pendingHeaderWithTermini types.PendingHeader) {
-	sl.phCache[pendingHeaderWithTermini.Termini[terminiIndex]] = pendingHeaderWithTermini
-}
-
-// pickPhCacheHead determines if the provided pendingHeader should be selected and returns true if selected
-func (sl *Slice) pickPhCacheHead(reorg bool, externPendingHeaderWithTermini types.PendingHeader, domOrigin bool) bool {
-	if reorg {
-		sl.pendingHeaderHeadHash = externPendingHeaderWithTermini.Termini[terminiIndex]
-		return true
-	}
-
-	localPendingHeader, exists := sl.phCache[sl.pendingHeaderHeadHash]
-
-	if domOrigin {
-		//calc local cache head reorg
-		localCacheReorg := true
-		for i := 0; i < common.NodeLocation.Context(); i++ {
-			localCacheReorg = (externPendingHeaderWithTermini.Header.NumberArray()[i].Cmp(localPendingHeader.Header.NumberArray()[i]) >= 0) && localCacheReorg
-		}
-
-		if exists && localCacheReorg && (externPendingHeaderWithTermini.Header.NumberU64() > localPendingHeader.Header.NumberU64()) {
-			return sl.updateCurrentPendingHeader(externPendingHeaderWithTermini)
-		}
-	}
-
-	return false
-}
-
 // updateCurrentPendingHeader compares the externPh parent td to the sl.pendingHeader parent td and sets sl.pendingHeader to the exterPh if the td is greater
-func (sl *Slice) updateCurrentPendingHeader(externPendingHeader types.PendingHeader) bool {
+func (sl *Slice) updatePendingHeader(externPendingHeader types.PendingHeader, local bool) bool {
```

Comment:
Is this function still supposed to update something? It looks like it just compares TDs and returns true/false. I don't see anything getting updated

Reply:
This function is no longer needed and has been removed.